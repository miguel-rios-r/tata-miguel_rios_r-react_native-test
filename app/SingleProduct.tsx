
import { Image, StyleSheet, Platform, TextInput, Alert, Pressable } from 'react-native';
import { useLocalSearchParams, useRouter } from 'expo-router';

import ParallaxScrollView from '@/components/ParallaxScrollView';
import { ThemedText } from '@/components/ThemedText';
import { ThemedView } from '@/components/ThemedView';
import { useEffect, useState } from 'react';
import SingleView from '@/components/SingleView';
import { useApiContext } from '@/context/ApiContext';

export default function SingleProduct() {

  const router = useRouter();  
  const { slug } = useLocalSearchParams();
  const { loading, currentProduct, deleteProduct, downloadProducts } = useApiContext();

  const releaseDate = new Date(currentProduct.date_release);
  const releaseDay = releaseDate.getDate();
  const releaseMonth = releaseDate.getMonth() + 1;
  const releaseYear = releaseDate.getFullYear();

  const revisionDate = new Date(currentProduct.date_revision);
  const revisionDay = revisionDate.getDate();
  const revisionMonth = revisionDate.getMonth() + 1;
  const revisionYear = revisionDate.getFullYear();
 
  return (
    <SingleView
      headerBackgroundColor={{ light: '#A1CEDC', dark: '#1D3D47' }}
      >
      {
        loading && 
        <ThemedView style={styles.stepContainer}>
          <ThemedText type="defaultSemiBold">Loading...</ThemedText>
        </ThemedView>
      }
      {
        !loading &&
        <ThemedView style={styles.stepContainer}>
          <ThemedText type='title'>ID: {currentProduct.id}</ThemedText>
          <ThemedText>Informaci贸n extra</ThemedText>
          <ThemedView style={styles.extraInfoContainer}>
            <ThemedText>Nombre</ThemedText>
            <ThemedText style={{width: 150}} type="defaultSemiBold">{currentProduct.name}</ThemedText>
          </ThemedView>
          <ThemedView style={styles.extraInfoContainer}>
            <ThemedText>Descripci贸n</ThemedText>
            <ThemedText style={{width: 150}} type="defaultSemiBold">{currentProduct.description}</ThemedText>
          </ThemedView>
          <ThemedText>Logo</ThemedText>
          <ThemedView style={{alignItems: "center"}}>
            <Image
              style={{width: "60%", height: 105,  marginBottom: 20}}
              source={{
                uri: currentProduct.logo,
              }}
            />
          </ThemedView>
          <ThemedView style={styles.extraInfoContainer}>
            <ThemedText>Fecha Liberaci贸n</ThemedText>
            <ThemedText style={{width: 150}} type="defaultSemiBold">{`${releaseDay}/${releaseMonth}/${releaseYear}`}</ThemedText>
          </ThemedView>
          <ThemedView style={styles.extraInfoContainer}>
            <ThemedText>Fecha Revisi贸n</ThemedText>
            <ThemedText style={{width: 150}} type="defaultSemiBold">{`${revisionDay}/${revisionMonth}/${revisionYear}`}</ThemedText>
          </ThemedView>
          
          <ThemedView style={{marginTop: 70}}>
            <Pressable style={{backgroundColor: "gray", padding: 10, alignItems: "center", borderRadius: 5, marginBottom: 5}} onPress={async () => {
              router.back();
              router.push("UpsertProduct")
            }}>
              <ThemedText style={{color: "white"}}>Editar</ThemedText>
            </Pressable>
            <Pressable style={{backgroundColor: "red", padding: 10, alignItems: "center", borderRadius: 5}} onPress={async () => {
              await deleteProduct(currentProduct.id);
              await downloadProducts();
              router.back();
            }}>
              <ThemedText style={{color: "white"}}>Eliminar</ThemedText>
            </Pressable>
          </ThemedView>
        </ThemedView>
      }
    </SingleView>
  );
}

const styles = StyleSheet.create({
  extraInfoContainer: {
    flexDirection: 'row',
    alignItems: 'flex-start',
    justifyContent: 'space-between',
  },
  stepContainer: {
    gap: 8,
    marginBottom: 8,
  }
});
