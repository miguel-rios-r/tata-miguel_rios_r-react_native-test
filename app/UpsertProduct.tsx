
import { StyleSheet } from 'react-native';

import { ThemedText } from '@/components/ThemedText';
import { ThemedView } from '@/components/ThemedView';
import SingleView from '@/components/SingleView';
import { ProductForm } from '@/components/ProductForm';
import { useApiContext } from '@/context/ApiContext';

export default function UpsertProduct() {

  const { loading, currentProduct } = useApiContext();

  return (
    <SingleView
      headerBackgroundColor={{ light: '#A1CEDC', dark: '#1D3D47' }}
      >
      {
        loading && 
        <ThemedView style={styles.stepContainer}>
          <ThemedText type="defaultSemiBold">Loading...</ThemedText>
        </ThemedView>
      }
      {
        !loading &&
        <>
          <ThemedView style={styles.titleContainer}>
            <ThemedText type="title">Formulario de registro</ThemedText>
          </ThemedView>
          {
            (currentProduct.id && currentProduct.id !== "") ? <ProductForm data={currentProduct}/> : <ProductForm/>
          }
        </>
        
      }
    </SingleView>
  );
}

const styles = StyleSheet.create({
  titleContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    gap: 8,
  },
  stepContainer: {
    gap: 8,
    marginBottom: 8,
  }
});
