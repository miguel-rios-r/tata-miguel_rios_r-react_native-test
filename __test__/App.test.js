import React from 'react';
import renderer from 'react-test-renderer';

import HomeScreen from '../app/(tabs)/index';
import SingleProduct from '../app/SingleProduct';
import UpsertProduct from '../app/UpsertProduct';

describe('<HomeScreen />', () => {
  it('has 1 child', () => {
    const tree = renderer.create(<HomeScreen />).toJSON();
    expect(tree.children.length).toBe(1);
  });
  it('renders correctly', () => {
    const tree = renderer.create(<HomeScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('<SingleProduct />', () => {
  it('has 1 child', () => {
    const tree = renderer.create(<SingleProduct />).toJSON();
    expect(tree.children.length).toBe(1);
  });
  it('renders correctly', () => {
    const tree = renderer.create(<SingleProduct />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('<UpsertProduct />', () => {
  it('has 1 child', () => {
    const tree = renderer.create(<UpsertProduct />).toJSON();
    expect(tree.children.length).toBe(1);
  });
  it('renders correctly', () => {
    const tree = renderer.create(<UpsertProduct />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
