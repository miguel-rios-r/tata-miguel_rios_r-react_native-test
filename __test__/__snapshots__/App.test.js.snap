// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      {
        "flex": 1,
      },
    ]
  }
>
  <RCTScrollView
    collapsable={false}
    forwardedRef={[Function]}
    jestAnimatedStyle={
      {
        "value": {},
      }
    }
    scrollEventThrottle={16}
  >
    <View>
      <View
        collapsable={false}
        jestAnimatedStyle={
          {
            "value": {
              "transform": [
                {
                  "translateY": 0,
                },
                {
                  "scale": 1,
                },
              ],
            },
          }
        }
        style={
          {
            "backgroundColor": "#A1CEDC",
            "height": 100,
            "overflow": "hidden",
            "transform": [
              {
                "translateY": 0,
              },
              {
                "scale": 1,
              },
            ],
          }
        }
      >
        <Image
          source={1}
          style={
            {
              "bottom": 0,
              "height": 100,
              "left": 0,
              "position": "absolute",
              "width": "100%",
            }
          }
        />
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#fff",
            },
            {
              "flex": 1,
              "gap": 16,
              "overflow": "hidden",
              "padding": 32,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "backgroundColor": "#fff",
              },
              {
                "alignItems": "center",
                "flexDirection": "row",
                "gap": 8,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#11181C",
                },
                undefined,
                {
                  "fontSize": 32,
                  "fontWeight": "bold",
                  "lineHeight": 32,
                },
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            Products
          </Text>
        </View>
        <RCTScrollView
          data={[]}
          getItem={[Function]}
          getItemCount={[Function]}
          keyExtractor={[Function]}
          onContentSizeChange={[Function]}
          onLayout={[Function]}
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={0.0001}
          stickyHeaderIndices={[]}
          viewabilityConfigCallbackPairs={[]}
        >
          <View />
        </RCTScrollView>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ffdc04",
              "borderRadius": 5,
              "marginTop": 40,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#11181C",
                },
                {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            Agregar
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`<SingleProduct /> renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      {
        "flex": 1,
      },
    ]
  }
>
  <RCTScrollView
    collapsable={false}
    forwardedRef={[Function]}
    jestAnimatedStyle={
      {
        "value": {},
      }
    }
    scrollEventThrottle={16}
  >
    <View>
      <View
        style={
          [
            {
              "backgroundColor": "#fff",
            },
            {
              "marginLeft": 24,
              "marginTop": 6,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "backgroundColor": "#fff",
              },
              {
                "gap": 8,
                "marginBottom": 8,
                "marginRight": 20,
                "marginTop": 20,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#11181C",
                },
                undefined,
                {
                  "fontSize": 32,
                  "fontWeight": "bold",
                  "lineHeight": 32,
                },
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            ID: 
          </Text>
          <Text
            style={
              [
                {
                  "color": "#11181C",
                },
                {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            Información extra
          </Text>
          <View
            style={
              [
                {
                  "backgroundColor": "#fff",
                },
                {
                  "alignItems": "flex-start",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#11181C",
                  },
                  {
                    "fontSize": 16,
                    "lineHeight": 24,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ]
              }
            >
              Nombre
            </Text>
            <Text
              style={
                [
                  {
                    "color": "#11181C",
                  },
                  undefined,
                  undefined,
                  {
                    "fontSize": 16,
                    "fontWeight": "600",
                    "lineHeight": 24,
                  },
                  undefined,
                  undefined,
                  {
                    "width": 150,
                  },
                ]
              }
            />
          </View>
          <View
            style={
              [
                {
                  "backgroundColor": "#fff",
                },
                {
                  "alignItems": "flex-start",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#11181C",
                  },
                  {
                    "fontSize": 16,
                    "lineHeight": 24,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ]
              }
            >
              Descripción
            </Text>
            <Text
              style={
                [
                  {
                    "color": "#11181C",
                  },
                  undefined,
                  undefined,
                  {
                    "fontSize": 16,
                    "fontWeight": "600",
                    "lineHeight": 24,
                  },
                  undefined,
                  undefined,
                  {
                    "width": 150,
                  },
                ]
              }
            />
          </View>
          <Text
            style={
              [
                {
                  "color": "#11181C",
                },
                {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            Logo
          </Text>
          <View
            style={
              [
                {
                  "backgroundColor": "#fff",
                },
                {
                  "alignItems": "center",
                },
              ]
            }
          >
            <Image
              source={
                {
                  "uri": "",
                }
              }
              style={
                {
                  "height": 105,
                  "marginBottom": 20,
                  "width": "60%",
                }
              }
            />
          </View>
          <View
            style={
              [
                {
                  "backgroundColor": "#fff",
                },
                {
                  "alignItems": "flex-start",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#11181C",
                  },
                  {
                    "fontSize": 16,
                    "lineHeight": 24,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ]
              }
            >
              Fecha Liberación
            </Text>
            <Text
              style={
                [
                  {
                    "color": "#11181C",
                  },
                  undefined,
                  undefined,
                  {
                    "fontSize": 16,
                    "fontWeight": "600",
                    "lineHeight": 24,
                  },
                  undefined,
                  undefined,
                  {
                    "width": 150,
                  },
                ]
              }
            >
              NaN/NaN/NaN
            </Text>
          </View>
          <View
            style={
              [
                {
                  "backgroundColor": "#fff",
                },
                {
                  "alignItems": "flex-start",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#11181C",
                  },
                  {
                    "fontSize": 16,
                    "lineHeight": 24,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ]
              }
            >
              Fecha Revisión
            </Text>
            <Text
              style={
                [
                  {
                    "color": "#11181C",
                  },
                  undefined,
                  undefined,
                  {
                    "fontSize": 16,
                    "fontWeight": "600",
                    "lineHeight": 24,
                  },
                  undefined,
                  undefined,
                  {
                    "width": 150,
                  },
                ]
              }
            >
              NaN/NaN/NaN
            </Text>
          </View>
          <View
            style={
              [
                {
                  "backgroundColor": "#fff",
                },
                {
                  "marginTop": 70,
                },
              ]
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "gray",
                  "borderRadius": 5,
                  "marginBottom": 5,
                  "padding": 10,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#11181C",
                    },
                    {
                      "fontSize": 16,
                      "lineHeight": 24,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    {
                      "color": "white",
                    },
                  ]
                }
              >
                Editar
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "red",
                  "borderRadius": 5,
                  "padding": 10,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#11181C",
                    },
                    {
                      "fontSize": 16,
                      "lineHeight": 24,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    {
                      "color": "white",
                    },
                  ]
                }
              >
                Eliminar
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`<UpsertProduct /> renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      {
        "flex": 1,
      },
    ]
  }
>
  <RCTScrollView
    collapsable={false}
    forwardedRef={[Function]}
    jestAnimatedStyle={
      {
        "value": {},
      }
    }
    scrollEventThrottle={16}
  >
    <View>
      <View
        style={
          [
            {
              "backgroundColor": "#fff",
            },
            {
              "marginLeft": 24,
              "marginTop": 6,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "backgroundColor": "#fff",
              },
              {
                "gap": 8,
                "marginBottom": 8,
                "marginRight": 20,
                "marginTop": 20,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#11181C",
                },
                undefined,
                {
                  "fontSize": 32,
                  "fontWeight": "bold",
                  "lineHeight": 32,
                },
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            Formulario de registro
          </Text>
        </View>
        <View
          style={
            [
              {
                "backgroundColor": "#fff",
              },
              {
                "gap": 8,
                "marginBottom": 8,
                "marginRight": 20,
                "marginTop": 20,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#11181C",
                },
                {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            ID
          </Text>
          <TextInput
            onChangeText={[Function]}
            style={
              {
                "borderColor": "#555",
                "borderRadius": 5,
                "borderWidth": 1,
                "height": 40,
                "marginBottom": 10,
                "padding": 10,
              }
            }
            value=""
          />
          <Text
            style={
              [
                {
                  "color": "#11181C",
                },
                {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            Nombre
          </Text>
          <TextInput
            onChangeText={[Function]}
            style={
              {
                "borderColor": "#555",
                "borderRadius": 5,
                "borderWidth": 1,
                "height": 40,
                "marginBottom": 10,
                "padding": 10,
              }
            }
            value=""
          />
          <Text
            style={
              [
                {
                  "color": "#11181C",
                },
                {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            Descripción
          </Text>
          <TextInput
            onChangeText={[Function]}
            style={
              {
                "borderColor": "#555",
                "borderRadius": 5,
                "borderWidth": 1,
                "height": 40,
                "marginBottom": 10,
                "padding": 10,
              }
            }
            value=""
          />
          <Text
            style={
              [
                {
                  "color": "#11181C",
                },
                {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            Logo
          </Text>
          <TextInput
            onChangeText={[Function]}
            style={
              {
                "borderColor": "#555",
                "borderRadius": 5,
                "borderWidth": 1,
                "height": 40,
                "marginBottom": 10,
                "padding": 10,
              }
            }
            value=""
          />
          <Text
            style={
              [
                {
                  "color": "#11181C",
                },
                {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            Fecha Liberación
          </Text>
          <View
            style={
              [
                {
                  "backgroundColor": "#fff",
                },
                {
                  "backgroundColor": "#F2F2F2",
                  "borderColor": "#F2F2F2",
                  "borderRadius": 5,
                  "borderWidth": 1,
                  "height": 40,
                  "marginBottom": 10,
                  "padding": 10,
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#11181C",
                  },
                  {
                    "fontSize": 16,
                    "lineHeight": 24,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ]
              }
            >
              20/6/2024
            </Text>
          </View>
          <Text
            style={
              [
                {
                  "color": "#11181C",
                },
                {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            Fecha Revisión
          </Text>
          <View
            style={
              [
                {
                  "backgroundColor": "#fff",
                },
                {
                  "backgroundColor": "#F2F2F2",
                  "borderColor": "#F2F2F2",
                  "borderRadius": 5,
                  "borderWidth": 1,
                  "height": 40,
                  "marginBottom": 10,
                  "padding": 10,
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#11181C",
                  },
                  {
                    "fontSize": 16,
                    "lineHeight": 24,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ]
              }
            >
              20/6/2025
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#ffdc04",
                "borderRadius": 5,
                "marginTop": 40,
                "padding": 10,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#11181C",
                  },
                  {
                    "fontSize": 16,
                    "lineHeight": 24,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ]
              }
            >
              Agregar
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#DDD",
                "borderRadius": 5,
                "padding": 10,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#11181C",
                  },
                  {
                    "fontSize": 16,
                    "lineHeight": 24,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ]
              }
            >
              Reiniciar
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
